velocity/src/main/java/me/leoko/advancedban/velocity/VelocityMethods.java

```diff
@@ -0,0 +1,368 @@
+import me.leoko.advancedban.manager.UUIDManager;
+import me.leoko.advancedban.utils.Punishment;
+import me.leoko.advancedban.utils.tabcompletion.TabCompleter;
+import me.leoko.advancedban.velocity.event.PunishmentEvent;
+import me.leoko.advancedban.velocity.event.RevokePunishmentEvent;
+import me.leoko.advancedban.velocity.listener.CommandReceiverVelocity;
+import net.kyori.text.TextComponent;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+public class VelocityMethods implements MethodInterface {
+
+    private final File configFile = new File(getDataFolder(), "config.yml");
+    private final File messageFile = new File(getDataFolder(), "Messages.yml");
+    private final File layoutFile = new File(getDataFolder(), "Layouts.yml");
+    private final File mysqlFile = new File(getDataFolder(), "MySQL.yml");
+    private Yaml config;
+    private Yaml messages;
+    private Yaml layouts;
+    private Yaml mysql;
+
+    @Override
+    public void loadFiles() {
+        config = LightningBuilder.fromFile(configFile).addInputStreamFromResource(configFile.getName()).setConfigSettings(ConfigSettings.PRESERVE_COMMENTS).createYaml();
+        messages = LightningBuilder.fromFile(messageFile).addInputStreamFromResource(messageFile.getName()).setConfigSettings(ConfigSettings.PRESERVE_COMMENTS).createYaml();
+        layouts = LightningBuilder.fromFile(layoutFile).addInputStreamFromResource(layoutFile.getName()).setConfigSettings(ConfigSettings.PRESERVE_COMMENTS).createYaml();
+        if (mysqlFile.exists()) {
+            mysql = LightningBuilder.fromFile(mysqlFile).setConfigSettings(ConfigSettings.PRESERVE_COMMENTS).createYaml();
+        } else mysql = config;
+    }
+
+    @Override
+    public String getFromUrlJson(String url, String key) {
+        try {
+            HttpURLConnection request = (HttpURLConnection) new URL(url).openConnection();
+            request.connect();
+
+            JsonParser jp = new JsonParser();
+            JsonObject json = (JsonObject) jp.parse(new InputStreamReader(request.getInputStream()));
+
+            String[] keys = key.split("\\|");
+            for (int i = 0; i < keys.length - 1; i++) {
+                json = json.getAsJsonObject(keys[i]);
+            }
+
+            return json.get(keys[keys.length - 1]).toString().replaceAll("\"", "");
+
+        } catch (Exception exc) {
+            return null;
+        }
+    }
+
+    @Override
+    public String getVersion() {
+        String version = "Unknown";
+        try (InputStream stream = getClass().getClassLoader().getResourceAsStream("velocity-plugin.json"); InputStreamReader read = new InputStreamReader(stream)) {
+            JsonObject obj = new JsonParser().parse(read).getAsJsonObject();
```

Comment:
There has to be a better way to do this. This method does IO and parses Json every time it is called. Why not use Velocity's methods on PluginContainer?

Reply:
Oh ok, my idea was different at begininng so I forgot to update this